### ISSUE TITLE
Compatibility with existing custom authentication

### ISSUE THREAD

**Issue by aronduby** [tymondesigns/jwt-auth#191]
I'm just diving into Laravel 5.1 and have created a custom authentication driver, following the steps at http://laravel.com/docs/5.1/authentication#adding-custom-authentication-drivers, which I'd like to use with this class now. I was hoping it would be quick and easy, but that doesn't seem to be the case (at least from a Laravel noob's standpoint). It would be sweet if it would just use whatever was setup in the auth config. Or maybe it can be made to do that but I'm just not seeing it.

**Comment by tymondesigns** [comment]
Firstly, is there a particular issue you're trying to solve by adding a custom auth driver?

You only need to satisfy the Auth contract [here](https://github.com/tymondesigns/jwt-auth/blob/master/src/Providers/Auth/AuthInterface.php) for it to work (which is laravels Auth by default)

**Comment by aronduby** [comment]
It looks like I got it figured out. I think I was getting pretty confused dealing with the Guard stuff.

Short version is I have some additional checks and an additional service provider that need to be involved for authentication, so I created my own driver. I was attempting to use that as the user provider for JWT which was failing miserably since I have some additional construction parameters.

What ended up working was keeping my custom driver setup like normal, but then create a new user provider for JWT similar to the EloquentUserAdapter.

Probably pretty obvious for people more versed in it, but the Guard thing not being well documented, and having a few things called User Provider threw me off.
